apps/api/pkg/handlers/signup.go

```diff
@@ -0,0 +1,164 @@
+package handlers
+
+import (
+	"api/pkg/database"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+)
+
+func SignUpHandler(w http.ResponseWriter, r *http.Request) {
+	// Read the request body
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, "Error reading request body", http.StatusInternalServerError)
+		return
+	}
+	defer r.Body.Close()
+
+	// Parse the request body as JSON
+	var requestData map[string]interface{}
+	if err := json.Unmarshal(body, &requestData); err != nil {
+		http.Error(w, "Error parsing request data", http.StatusBadRequest)
+		return
+	}
+
+	// Extract email, password, firstName, and lastName from the request data
+	email, emailExists := requestData["email"].(string)
+	password, passwordExists := requestData["password"].(string)
+	firstName, firstNameExists := requestData["firstName"].(string)
+	lastName, lastNameExists := requestData["lastName"].(string)
+
+	// Check if the required fields exist
+	if !emailExists || !passwordExists || !firstNameExists || !lastNameExists {
+		http.Error(w, "Invalid request format", http.StatusBadRequest)
+		return
+	}
+
+	// Construct the JSON payload for Auth0 API
+	data := map[string]interface{}{
+		"email":          email,
+		"user_metadata":  map[string]interface{}{},
+		"blocked":        false,
+		"email_verified": false,
+		"app_metadata":   map[string]interface{}{},
+		"given_name":     firstName,
+		"family_name":    lastName,
+		"name":           firstName + " " + lastName,
+		"connection":     "Username-Password-Authentication",
+		"password":       password,
+		"verify_email":   false,
```

Comment:
verify_email and email_verified looks like same thing. Why we have almost same keys? Is auth0 want these values?